tsuru/client/apps.go

```diff
@@ -1241,3 +1241,117 @@ func (c *UnitRemove) Run(context *cmd.Context, client *cmd.Client) error {
 	}
 	return cmd.StreamJSONResponse(context.Stdout, response)
 }
+
+type UnitSet struct {
+	cmd.GuessingCommand
+	fs *gnuflag.FlagSet
+}
+
+func (c *UnitSet) Info() *cmd.Info {
+	return &cmd.Info{
+		Name:  "unit-set",
+		Usage: "unit-set <process name> <# of units> [-a/--app appname]",
```

Comment:
I think the process name could be in a `-p/--process` flag, following the same pattern from `unit-add` and `unit-remove` commands.

Reply:
I thought about that, but in `unit-add` and `unit-remove` the `-p/--process` flag is optional, and without specifying it, the change is applied to all processes.

In `unit-set`, if the `-p/--process` flag is not informed, I'm unsure about what to do, specially if I have a scenario like:

- `web` process with 1 unit
- `worker1` process with 5 units
- `worker2` process with 3 units

In this case, if I run `unit-set` with `3`, it should:

- add 2 units to `web`
- remove 2 units from `worker1`
- don't change anything for `worker2`

It seemed to me to be a little cumbersome, even to inform the user what is being performed, that's why I went with the current approach. But if you prefer, I can change it 😄 

If we can use `-p/--process` as a flag and make it mandatory, that seems to be a good alternative. I just can't remember right now a Tsuru command that follows this pattern (mandatory flag). If you decide for this option, would be helpful to point a command with this pattern, so I can use as a starting point.

Reply:
You could make the `-p/--process` flag optional when the app has only one process, and require it for apps with more processes.

We use this pattern in `app create` command. The `--team` flag is optional when the user is part of only one team, and required otherwise.

Reply:
I like the idea. But I took a look in `app create` and that check is actually done on the API (https://github.com/tsuru/tsuru/blob/master/api/app.go#L291).

Would you like me to do the same here? I'm happy with doing it if you think it's a good idea.

Reply:
Maybe in this case we should do it in the client, because there isn't a specific route for `unit-set` command.

Reply:
Nice, I will make the changes over the next few days

Reply:
@ggarnier Done, could you check again please?